docs/griptape-framework/data/image-generation-engines.md

```diff
@@ -0,0 +1,147 @@
+## Overview
+
+Image generation engines facilitate the use of [image generation drivers](../structures/image-generation-drivers.md) by image generation tasks and tools. Each image generation engine defines a `run` method that accepts the inputs necessary for each image generation mode, combines these inputs with any available rulesets, and provides the request to the configured image generation driver.
+
+#### Rulesets
+
+[Rulesets](../structures/rulesets.md) provided to image generation engines are combined with prompts, providing further instruction to image generation models. In addition to typical Rulesets, image generation engines support Negative Rulesets. Negative Rulesets are used by [image generation drivers](../structures/image-generation-drivers.md) with support for prompt wieghting and used to influence the image generation model to avoid undesireable features described by negative prompts.
+
+### Prompt Image Generation Engine 
+
+This image generation engine facilitates generating images from text prompts.
+
+```python
+from griptape.structures import Agent
+from griptape.engines import PromptImageGenerationEngine
+from griptape.drivers import AmazonBedrockImageGenerationDriver, \
+    BedrockStableDiffusionImageGenerationModelDriver
+from griptape.tools import PromptImageGenerationClient
+
+
+# Define positive and negative rulesets.
+positive_ruleset = Ruleset(rules=[Rule("realistic"), Rule("high quality")])
+negative_ruleset = Ruleset(rules=[Rule("distorted")])
+
+# Create a driver configured to use Stable Diffusion via Bedrock.
+driver = AmazonBedrockImageGenerationDriver(
+    image_generation_model_driver=BedrockStableDiffusionImageGenerationModelDriver(),
+    model="stability.stable-diffusion-xl-v0",
+)
+
+# Create an engine configured to use the driver.
+engine = PromptImageGenerationEngine(
+    rulesets=[positive_ruleset],
+    negative_rulesets=[negative_ruleset],
```

Comment:
this is a lot of code, which means a lot to maintain if we make refactors or upstream changes. Are we able to automate testing it? Should we pare it down to only a handful of lines?

Reply:
We currently do automate testing this, see `tests/integration/test_code_snippets.py`. Unfortunately that means we need the boilerplate dependency instantiation because this is real code that gets executed.

Reply:
@andrewfrench can you try creating a `tests/assets/` directory to see if the code snippets can pull resources from there?

Reply:
done! The LLM looks happy to pull from there.